File: tests/property/python/areas/arith_extended.py
"""SageMath side of the ``arith_extended`` property-test area.

Cases: tests/property/cases/arith_extended.cases.json
"""

from sage.all import bernoulli, binomial, fibonacci


def bernoulli_numerator(n):
    """Return the exact numerator of ``B_n``."""
    return bernoulli(n).numerator()


def bernoulli_denominator(n):
    """Return the positive exact denominator of ``B_n``."""
    return bernoulli(n).denominator()


FUNCTIONS = {
    'binomial': binomial,
    'fibonacci': fibonacci,
    'bernoulli_numerator': bernoulli_numerator,
    'bernoulli_denominator': bernoulli_denominator,
}

File: tests/property/typescript/areas/arith_extended.ts
/**
 * sagemath-ts side of the `arith_extended` property-test area.
 *
 * Cases: tests/property/cases/arith_extended.cases.json
 * SageMath counterpart: tests/property/python/areas/arith_extended.py
 */

import { bernoulli, binomial, fibonacci } from '../../../../packages/sagemath-ts/src/arith/misc.js';

export const functions = {
  binomial: (n: bigint, k: bigint): bigint => binomial(n, k),
  fibonacci: (n: bigint): bigint => fibonacci(n),
  bernoulli_numerator: (n: bigint): bigint => bernoulli(n).numerator,
  bernoulli_denominator: (n: bigint): bigint => bernoulli(n).denominator,
};
